python/run-tests.py

```diff
@@ -160,11 +164,33 @@ def run_individual_python_test(target_dir, test_name, pyspark_python):
 
 
 def get_default_python_executables():
-    python_execs = [x for x in ["python2.7", "python3.6", "pypy"] if which(x)]
-    if "python2.7" not in python_execs:
-        LOGGER.warning("Not testing against `python2.7` because it could not be found; falling"
-                       " back to `python` instead")
-        python_execs.insert(0, "python")
+    python_execs = [x for x in ["python3.6", "pypy"] if which(x)]
+
+    if "python3.6" not in python_execs:
+        p = which("python3")
+
+        if p:
+            py_out = subprocess.run([p, "--version"], stderr=subprocess.PIPE)
+            py_out = py_out.stderr.decode("utf-8")
```

Comment:
why it is stderr? I ran with this change, stdout works but stderr not.

Reply:
hmm in my testing it was STDERR...

Reply:
It might be easier to check the version in the subprocess and then return a non-zero if it is the wrong version.  so something like this:
```python
python -c \
'''
import sys
from distutils.version import LooseVersion
if LooseVersion(sys.version) < LooseVersion('3.6.0'):
    sys.exit(1)
'''
```
Then just check the return code of the process

Reply:
ah word.  sgtm